driver-core/src/test/java/com/datastax/driver/core/ClusterAssert.java

```diff
@@ -49,6 +50,13 @@ public HostAssert host(int hostNumber) {
         return new HostAssert(host, actual);
     }
 
+    public HostAssert host(String hostAddress) {
```

Comment:
Couldn't this be refactored into a new method `TestUtils.findHost(Cluster, String)` so that `TestUtils.findHost(Cluster, int)` could call it? Also, the same caveat probably applies here: we cannot make assertions on a host that hasn't joined the cluster yet or that has left it.